File: shared/Utilities/DurationFormatter.swift
import Foundation

public enum DurationFormatter {
    public static func elapsed(_ interval: TimeInterval) -> String {
        let total = max(0, Int(interval))
        let hours = total / 3600
        let minutes = (total % 3600) / 60
        let seconds = total % 60
        if hours > 0 {
            return String(format: "%d:%02d:%02d", hours, minutes, seconds)
        }
        return String(format: "%d min", max(1, minutes + (seconds > 0 && minutes == 0 ? 1 : 0)))
    }

    public static func detailed(_ interval: TimeInterval) -> String {
        let total = max(0, Int(interval))
        let hours = total / 3600
        let minutes = (total % 3600) / 60
        let seconds = total % 60
        if hours > 0 {
            return String(format: "%dh %dm", hours, minutes)
        }
        if minutes > 0 {
            return String(format: "%d min", minutes)
        }
        return String(format: "%d sec", seconds)
    }
}

File: android/domain/src/main/kotlin/com/padelscore/domain/DurationFormatter.kt
package com.padelscore.domain

object DurationFormatter {
    fun countdown(interval: Double): String {
        val total = maxOf(0, interval.toInt())
        val minutes = total / 60
        val seconds = total % 60
        return "%d:%02d".format(minutes, seconds)
    }

    fun elapsed(interval: Double): String {
        val total = maxOf(0, interval.toInt())
        val hours = total / 3600
        val minutes = (total % 3600) / 60
        val seconds = total % 60
        if (hours > 0) {
            return "%d:%02d:%02d".format(hours, minutes, seconds)
        }
        val shownMinutes = maxOf(1, minutes + if (seconds > 0 && minutes == 0) 1 else 0)
        return "$shownMinutes min"
    }

    fun detailed(interval: Double): String {
        val total = maxOf(0, interval.toInt())
        val hours = total / 3600
        val minutes = (total % 3600) / 60
        val seconds = total % 60
        return when {
            hours > 0 -> "${hours}h ${minutes}m"
            minutes > 0 -> "$minutes min"
            else -> "$seconds sec"
        }
    }
}
